### ISSUE TITLE
default_interpolation -> geometric_interpolation ? 

### ISSUE THREAD

**Issue by KnutAM** [Ferrite-FEM/Ferrite.jl#950]
https://github.com/Ferrite-FEM/Ferrite.jl/blob/41cfff526912c64ebe41db83cbe7315e454d6660/src/Grid/grid.jl#L134-L139

To me this function name doesn't reflect the docstring (which describes what the function actually does IMO).
Wouldn't it be better to call this `geometric_interpolation`, otherwise I find it confusing as `default` indicates that there are other alternative interpolations for a given cell? 

Or are there any cases that I'm not aware of when different geometric interpolations can be used for the same cell? If not, I'd suggest changing this in 1.0. 

(The old name could be kept to avoid making things more breaking for users, but perhaps removed from API. This would then be to just get a default for the function interpolation on a cell, which I guess the function was originally intended for).

**Comment by KnutAM** [comment]
When? 
```
julia> using Ferrite
julia> grid = generate_grid(QuadraticQuadrilateral, (2,2));
julia> cv = CellValues(QuadratureRule{RefQuadrilateral}(2), Lagrange{RefQuadrilateral,1}(), Lagrange{RefQuadrilateral,1}());

julia> x = getcoordinates(grid, 1);

julia> reinit!(cv, x)
ERROR: ArgumentError: the number of (geometric) base functions (4) does not match the number of coordinates in the vector (9). Perhaps you forgot to use an appropriate geometric interpolation when creating FE values? See https://github.com/Ferrite-FEM/Ferrite.jl/issues/265 for more details.
```
Or is there another context you refer to?

**Comment by termi-official** [comment]
> https://github.com/Ferrite-FEM/Ferrite.jl/blob/41cfff526912c64ebe41db83cbe7315e454d6660/src/Grid/grid.jl#L134-L139
> 
> To me this function name doesn't reflect the docstring (which describes what the function actually does IMO). Wouldn't it be better to call this `geometric_interpolation`, otherwise I find it confusing as `default` indicates that there are other alternative interpolations for a given cell?
> 
> Or are there any cases that I'm not aware of when different geometric interpolations can be used for the same cell? If not, I'd suggest changing this in 1.0.
> 
> (The old name could be kept to avoid making things more breaking for users, but perhaps removed from API. This would then be to just get a default for the function interpolation on a cell, which I guess the function was originally intended for).

Since we are already breaking the interfaces I second the change. We use `default_interpolation` internally to query the correct geometric for cells. I would even go one step further and make this function part of the public API, as I also find myself using this one from time to time (e.g. for visualization).

And then I would go another step further and like to add the invariant to the docs that cells should come with **exactly one** interpolation, which is the return value of `geoemtric_interpolation`, since we cannot guarantee consistency between operations otherwise. Having the possibility that cells might come with multiple interpolations equipped is quite a headache for me at least.

**Comment by lijas** [comment]
I am fine with name changes like this, but I have been using Ferrite.default_interpolation quite a lot in my research cod to setup CellValues independent of the cell type (and I guess others have too). So I would suggest to also "soft-deprecating" it e.g. `const default_interpolation = geometric_interpolation` or use `@deprecate`, so users don't have to bother updating "annoying" name changes like these. I don't know if that is considered bad practice though.

**Comment by KnutAM** [comment]
I agree with both of you! Will make PR then later today :)

**Comment by fredrikekre** [comment]
Fixed by #953 but could use a changelog entry since it is a new export.
